sw/host/opentitanlib/src/otp/alert_handler.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::otp::alert_handler_regs::*;
+use crate::otp::lc_state::LcStateVal;
+use crate::otp::otp_img::OtpRead;
+
+use anyhow::{bail, Result};
+use bitvec::prelude::*;
+use crc::{Crc, Digest};
+use num_enum::TryFromPrimitive;
+
+use std::convert::TryFrom;
+
+#[derive(Clone, Copy)]
+struct AlertClassRegs {
+    regwen: u32,
+    ctrl: u32,
+    accum_thresh: u32,
+    timeout_cyc: u32,
+    phase_cycs: [u32; ALERT_HANDLER_PARAM_N_PHASES as usize],
+}
+
+pub struct AlertRegs {
+    /// ALERT_HANDLER_LOC_ALERT_REGWEN
+    regwen: [u32; ALERT_HANDLER_LOC_ALERT_REGWEN_MULTIREG_COUNT as usize],
+    /// ALERT_HANDLER_ALERT_EN_SHADOWED
+    en: [u32; ALERT_HANDLER_ALERT_EN_SHADOWED_MULTIREG_COUNT as usize],
+    /// ALERT_HANDLER_ALERT_CLASS_SHADOWED
+    class: [u32; ALERT_HANDLER_ALERT_CLASS_SHADOWED_MULTIREG_COUNT as usize],
+    /// ALERT_HANDLER_LOC_ALERT_REGWEN
+    loc_regwen: [u32; ALERT_HANDLER_LOC_ALERT_REGWEN_MULTIREG_COUNT as usize],
+    /// ALERT_HANDLER_LOC_ALERT_EN_SHADOWED
+    loc_en: [u32; ALERT_HANDLER_LOC_ALERT_EN_SHADOWED_MULTIREG_COUNT as usize],
+    /// ALERT_HANDLER_LOC_ALERT_CLASS_SHADOWED
+    loc_class: [u32; ALERT_HANDLER_LOC_ALERT_CLASS_SHADOWED_MULTIREG_COUNT as usize],
+    /// Alert handler class registers
+    class_regs: [AlertClassRegs; ALERT_HANDLER_PARAM_N_CLASSES as usize],
+}
+
+#[derive(TryFromPrimitive)]
```

Comment:
We should use bindgen for these constants too.

Reply:
That being said, we can address this in a follow-up, I think @cfrantz has a PR for this.

Reply:
I agree, looks like https://github.com/lowRISC/opentitan/pull/14860 covers this; will need to unify these

Reply:
Just a quick note: #14860 was rolled-back in 28f79af3c. @timothytrippel do we track its return to the stage somewhere?